lib/logstash/event.rb

```diff
@@ -295,11 +295,7 @@ def init_timestamp(o)
 
   public
   def to_hash_with_metadata
-    if @metadata.nil?
-      to_hash
-    else
-      to_hash.merge(METADATA => @metadata)
-    end
+    to_hash.merge(METADATA => @metadata)
```

Comment:
re: linked code.

Well the intent is to provide a fall back if no metadata key is supplied at object creation time, we never enforce it to be a hash. (maybe we should?)

The only place I could see a benefit to return nil is if we were serializing that field... but we dont serialize metadata.